src/tests/ftest/control/dmg_pool_query_ranks.py

```diff
@@ -141,8 +143,21 @@ def test_pool_query_ranks_mgmt(self):
         self.log.info("Starting reintegrating ranks: all_ranks=%s", all_ranks)
         for rank in all_ranks:
             self.log.debug("Reintegrating rank %d", rank)
-            self.pool.reintegrate(rank)
 
+            counter = 0
+            cmd_succeed = False
+            while counter < 3:
+                try:
+                    result = self.pool.reintegrate(rank)
+                except CommandFailure:
+                    self.log.debug("dmg command failed retry")
+                else:
+                    cmd_succeed = True
+                    break
+                counter += 1
+                time.sleep(3)
+
+            self.assertTrue(cmd_succeed, result)
```

Comment:
We should print a more descriptive failure message. For example
```suggestion
            self.assertTrue(cmd_succeed, "pool reintegrate failed: {}".format(result))
```